src/Agent.Worker/Util/ExceptionsUtil.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Sockets;
+using System.Text;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Agent.Worker.Util
+{
+    public class ExceptionsUtil
+    {
+        public static void HandleSocketException(SocketException e, string url, IExecutionContext trace)
+        {
+            trace.Error("SocketException occurred.");
```

Comment:
Please avoid duplication - you can probably use some pattern like simple strategy or pass trace method as delegate here.

Reply:
Earlier I had issue with passing different types of tracers to handler, so several handlers were created as solution for different types of tracers. Currently I have 1 handler in SDK for all types of tracers. And it's solved.